src/shape.rs

```diff
@@ -256,18 +256,10 @@ impl<'a> From<&'a swf::tags::DefineShape> for Shape<'a> {
 
                     continue;
                 }
-                swf::ShapeRecord::StraightEdge(line) => Line {
+                swf::ShapeRecord::Edge(edge) => Line {
                     from: Point::default(),
-                    bezier_control: None,
-                    to: Point::from(&line.delta),
-                },
-                swf::ShapeRecord::CurvedEdge(bezier) => {
-                    let control = Point::from(&bezier.control_delta);
-                    Line {
-                        from: Point::default(),
-                        bezier_control: Some(control),
-                        to: control + Point::from(&bezier.anchor_delta),
-                    }
+                    bezier_control: edge.control_delta.map(|control| Point::from(&control)),
+                    to: Point::from(&edge.delta),
```

Comment:
Nice! Oh, also, `impl<'a> From<&'a swf::Vector2D> for Point` can be just `impl From<swf::Vector2D> for Point` now that `swf::Vector2D` is `Copy` (if you don't want to bother, I can merge the PR as-is and make that change later myself)

Reply:
Fixed in 30bfd25713d29ac3fde970e76b8789cd59c42344